### ISSUE TITLE
[Website, Label, Alerts, Links] Background color and text color mixin and classes

### ISSUE THREAD

**Issue by joshbruce** [uswds/uswds#1242]
## Issue type
- Website
- Label
- Alerts
- Amendment to #1158 
- Groundwork for #1239 
## Description
1. The compiled CSS does not apply a color for text or links based on background color. 
2. Currently, for the "Visual style" page, we have helpers in applying background colors, as well as font colors using two separate class definitions (one for background and one for text color - see "Neutrals on colored backgrounds). Promoting these classes to a new `_colors.scss` file would allow defining font and link colors based on background color (note: transparent and white should use defaults), also allows developers to apply background colors without modifying or extending the compiled CSS. 
3. Making a `@mixin` within the `_utilities.scss` would also allow those extending the UI Components to apply background colors easily. We also don't have an example of links on a background other than white or transparent - some backgrounds could or would result in a reduction in contrast; reducing accessibility considerations.
## Steps to reproduce the issue
1. View `/docs/doc_assets/css/styleguide.scss` line 592 to see the background color helpers.
2. See #1239 to see how this could be leveraged on labels.
3. View "Alerts" page to see how it could be leveraged for the alerts component.
4. View "Buttons" to see how this might be used there as well - possibly not recommended because, semantically, interactive content (links) cannot be inside a `button`; therefore, use of the classes or `@mixin` would increase code bloat with no benefit.
## Other
- [ ] We don't have an example of `color-primary-alt-light` should we? If so, should it use base or white?
- [ ] We don't have an example of `color-secondary-light` should we? If so, should it use base or white?
- [ ] We don't have an example of `color-gray-lightest` should we? If so, should it use base or white?
- [ ] We don't have an example of `color-gray-warm-dark` should we? If so, should it use base or white?
- [ ] We don't have an example of `color-gray-cool-light` should we? If so, should it use base or white?
- [ ] We don't have an example of `color-cool-blue` should we? If so, should it use base or white?

**Comment by zeinamakky** [comment]
I will try to work on this.

**Comment by maya** [comment]
@joshbruce thanks for filing this issue! Agree that it would be helpful to add in some background color helpers to the standards. The benefit would be more for people to prototype things quickly so would not recommend it for production code in most cases.

**Comment by joshbruce** [comment]
@maya - Agreed. I try to "pair design" as much as possible - having a designer paired with a developer; so, I'm constantly thinking of ways to allow developers to quickly change the look and feel without having to switch out of the HTML. In my experience, something like this - and other classes - can be used to quickly change things based on designer (or user) feedback, in the moment, with a more consolidated definition created from it.

Where we went with #1262, my mind is starting to work on an alternative (addition...??) to the #1244 implementation; though I do appreciate the flexibility (affordances) offered by the `@mixin`.

**Comment by joshbruce** [comment]
@maya - Would you like me to close this issue?

See also comment on #1244

**Comment by maya** [comment]
@joshbruce I think it's fine to leave open for now, thanks!

**Comment by maya** [comment]
Closing as this is being covered in the utilities work. Thanks!
